### PR TITLE
Bytecomp/Dll: cleanup API

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by xavierleroy** [comment]
@hhugo : you have pushed more than 50 times to ocaml/ocaml in the last week, meaning 50+ mails in my mailbox.  I'm seriously considering unsubscribing from ocaml/ocaml notifications, which may not be good for the future of OCaml, just because you're making a lot of noise there.

Could you please at least not submit PRs in the Draft state and just work on them in your private repo? 

 Also, nobody asked you to do all this "cleanup" to your taste, so it would be wise to ask whether your PRs in this direction are welcome before submitting them in passive-aggressive style.

**Comment by hhugo** [comment]
> you have pushed more than 50 times to ocaml/ocaml in the last week, meaning 50+ mails in my mailbox. I'm seriously considering unsubscribing from ocaml/ocaml notifications.

To be precise, I've pushed many times to my own fork of the ocaml repo. Many of the commits were "linked" to open draft PRs on ocaml/ocaml. Github notifications on draft PRs seems to be a known source of annoyance (https://github.com/community/community/discussions/10409). Maybe `CONTRIBUTING.md` should say something about how to use the draft feature. That said, I'll stop using draft PRs.

>  Also, nobody asked you to do all this "cleanup" to your taste, so it would be wise to ask whether your PRs in this direction are welcome before submitting them in passive-aggressive style.

I don't known where you get the passive-aggressive style from. I'm just trying to contribute changes that I believe make the codebase easier to understand and review.

This very PR (while being draft) seems to go in the direction you 've suggested in https://github.com/ocaml/ocaml/pull/9551#pullrequestreview-442695491.

**Comment by xavierleroy** [comment]
> Many of the commits were "linked" to open draft PRs on ocaml/ocaml.

Indeed.  Would you please not open a PR until you're happy with your code and have no further pushes to make before review?  While you're at it, please add descriptions to your PR explaining what problem you're trying to solve.

**Comment by gasche** [comment]
@hhugo I think it would be worth having a higher-level explanation of the motivation for your work. I can hazard the guess that you are trying to make compiler-libs easier to use, but maybe that's something elsee. In any case, seeing the big picture would create the space for a big-picture discussion (what are the costs/benefits of the proposed changes), allow people to express high-level support (or lack thereof), and in particular seeing if and when we want to spend the reviewing resources on the work.

If people like the high-level idea and you have an idea of who is interested in pouring in the reviewing work, then maybe you could go at your PR more sequentially -- one or a couple PR at a time rather than all together -- to spread the work and activity.

To summarize, playing the "good cop" part of the duo: could you write down a "motivation" or "roadmap" document that explains why we should care about your work?

(Note: "it makes the code nicer" is an argument but one should use carefully because, in addition to "nicer" sometimes being subjective (but often not), there is a real risk of introducing regressions, and the maintainer-side cost in term of human resources that are not spent on something else. On the positive side, personally I thing it's useful when more people grow familiar with a piece of code, and that can happen through those refactorings; if you want to sell me on this, think of also adding documentation comments where they would be useful.)
